airflow/providers/google/cloud/operators/automl.py

```diff
@@ -219,7 +246,11 @@ def __init__(
     ) -> None:
         super().__init__(**kwargs)
 
+        if model_id is None and endpoint_id is None:
+            raise AirflowException("You must specify model_id or endpoint_id!")
```

Comment:
1. `model_id` is a template field; related value checks shouldn't be here - but in the beginning of the `execute()` method. Tou could relocate this `if` there, while referring to `model_Id` as `self.model_id` and `endpoint_id` as `self.endpoint_id`.
2. Please add a test for this `if` after relocating it.